#!/usr/bin/env python
PACKAGE = "vastar_planner"

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, int_t, double_t, bool_t

gen = ParameterGenerator()

gen.add("threshold", int_t, 0, "only search space less than the value", 228, 1, 255)
gen.add("stop_deviance", int_t, 0, "remove last n points in path", 0, 0, 20)
gen.add("planner_weight_path", double_t, 0, "planner weight of path", 4.0, 1.0, 50.0)
gen.add("planner_weight_obs", double_t, 0, "planner weight of costmap", 1.0, 1.0, 50.0)
gen.add("smooth_weight", double_t, 0, "smooth weight", 0.1, 0.0, 0.4)
gen.add("smooth_times", int_t, 0, "smooth smooth_times", 2, 1, 10)
gen.add("allow_vague_search", bool_t, 0, "allow_vague_search", False)
gen.add("use_waypoint", bool_t, 0, "wether to search waypoint insead of goal point.", False)
exit(gen.generate(PACKAGE, "vastar_planner", "VastarPlanner"))
